nestkernel/target_data.h

```diff
@@ -163,16 +181,21 @@ class TargetData
   // and to handle variant fields
 
 private:
-  static const unsigned int default_marker_ = 0;
-  static const unsigned int complete_marker_ = 1;
-  static const unsigned int end_marker_ = 2;
-  static const unsigned int invalid_marker_ = 3;
 
-  unsigned int source_lid_ : 19; //!< local id of presynaptic neuron
-  unsigned int source_tid_ : 10; //!< thread index of presynaptic neuron
-  unsigned int marker_ : 2;
-  bool is_primary_ : 1; //!< TargetData has TargetDataFields
-                        //!< else has SecondaryTargetDataFields
+  static constexpr uint8_t NUM_BITS_LID = 19U;
+  // NUM_BITS_TID set via cmake
+  static constexpr uint8_t NUM_BITS_MARKER = 2U;
+  static constexpr uint8_t NUM_BITS_IS_PRIMARY = 1U;
+
+  static constexpr int MAX_LID = generate_max_value( NUM_BITS_LID );
+  static constexpr int MAX_MARKER = generate_max_value( NUM_BITS_MARKER );
+
+  unsigned int source_lid_ : NUM_BITS_LID; //!< local id of presynaptic neuron
+  //!< thread index of presynaptic neuron
+  unsigned int source_tid_:  NUM_BITS_TID;
+  unsigned int marker_ : NUM_BITS_MARKER;
+   //!< TargetData has TargetDataFields else SecondaryTargetDataFields
```

Comment:
Doxygen documentation with `//!<` points to the variable that precedes it, so these should be `//!` only.